entry-points/spin-entry-point.js

```diff
@@ -55,19 +63,60 @@ class SpinderUi extends Component { {
 
     if (validaMatchRankings) {
       validaMatchRankings.sort();
-      return validMatchRankings.get(0);
+      return validMatchRankings.get(0)
     }
 
     return ProfileData.botUser;
   },
 
-  onSpinEvent(event) {
-    /* TODO */
+  onSpinEvent(matchLevel) {
+    const { prefTimeOfDay, age, region } = this.props;
+    const currentMatch = this.state.currentMatch;
+    const reciepientResponce = ProfileData.sendMatchLevel(this.props.username, currentMatch, matchLevel);
+
+    if (reciepientResponce) {
+      this.setState({openMessenger: true});
+    } else {
+      /* If current match has not yet responded, move on to the next match */
+      this.setState({currentMatch: generateMatch(prefTimeOfDay, age, region)});
+    }
+  },
+
+  closeMessenger() {
+    this.setState({
+      openMessenger: false
+    });
   },
 
   render() {
-    /* TODO */
+    if (initialLoad) {
+      return (
+        <View>
+          <Header>
+            Wave Your Phone Around If You Are Looking For Love!
+            <Spinder mode='initial' beginMatching={() => this.onSpinEvent}/>
+          </Header>
+        </View>
+      )
+    }
+    if (openMessenger) {
+      return (
```

Comment:
I few have the messenger logic here we should consider creating new components to hold the logic for hosting the spinner and the messenger. We should discuss this next time in planning.